github/GithubException.py

```diff
@@ -106,6 +106,57 @@ def __str__(self) -> str:
 class BadCredentialsException(GithubException):
     """
     Exception raised in case of bad credentials (when Github API replies with a 401 or 403 HTML status)
+
+    ### Possible Causes
+
+    This exception is raised possibly because of:
+    1. Insufficient scope of token
+    2. An expired token. In an application with **cache** mechanism, the cached PyGithub-like objects might
+    contain an expired token unexpectedly. ATTENTION: The PyGithub-like objects can not update the token
+    automatically.
+
+    ### One way to find the cause
+    
+    One way to find the root cause is as follows.
+    The steps are abstracted from https://github.com/PyGithub/PyGithub/issues/1753#issuecomment-1039533679
```

Comment:
```suggestion
    ### Expose the token in the logs

    Attention: This is for debugging purposes only, as this leaks the token into log files, which is a security risk.

    The best way to investigate the root cause is to show the actual token used to authenticate in the logs.
    Usually, the token is obfuscated for security reasons. Here we deliberately write the plain token to the logs,
    so that we can compare it with the expected token.
```